simplesam.py

```diff
@@ -42,32 +42,21 @@ class GenomicOrder(object):
     def __gt__(self, other):
         if self.rname != other.rname:
             return self.rname > other.rname
-        elif self.pos != other.pos:
-            return self.pos > other.pos
-        else:
-            return str(self) > str(other)
+        return self.pos > other.pos
```

Comment:
I'd like to leave the string comparison. It might not be obvious, but `GenomicOrder` is only inherited by `Sam`, and the purpose of the string comparison is to make sorting order deterministic in case of `(chrom, pos)` ties. If you can come up with a better method of breaking sorting ties for SAM records then I'm open to it.

Reply:
Oh, maybe I'm missing a subtlety: is string comparison implemented for `object` and somehow default even if you update __cmp__ methods?

Reply:
My issue here was that, by comparing the string versions of the objects, I think you'll be taking into account memory address in your comparison, which is effectively random. The removal of that line was specifically because of wanting the string comparison to be deterministic. I honestly am not the most knowledgeable about the details of SAM/BAM; are there other features about the files that can be added here to capture deterministic behavior?